sdk/core/scheduler/timer.h

```diff
@@ -72,6 +117,39 @@ namespace
 					break;
 				}
 			}
+			if (!Thread::any_ready())
+			{
+				for (Thread *iter = Thread::waitingList; iter;)
+				{
+					if (iter->is_yielding())
+					{
+						Debug::log("Woke thread {} {} cycles early",
+						           iter->id_get(),
+						           int64_t(iter->expiryTime) - now);
+						Thread *iterNext = iter->timerNext;
+						iter->ready(Thread::WakeReason::Timer);
+						iter = iterNext;
+						if (Thread::waitingList == nullptr ||
+						    iter == Thread::waitingList)
+						{
+							break;
+						}
+					}
+					else
+					{
+						break;
```

Comment:
I don't think that's right; I think you want to keep scanning the queue for the first yielding thread, not bail if the head of the queue isn't yielding.

Reply:
This probably needs a comment.  I actually do (and therefore probably don't want to write this as a loop).  There's a fun case of:

 - Low-priority B sleeps 3 ticks.
 - High-priority A wakes, runs, and yields 5 ticks.

I think I want B to wake when it finishes sleeping, but then A to wake when B has finished.  

I'm not 100% sure about this, but there are a lot of things similar to this (e.g. B ties to acquire a lock with a 1-tick timeout in the middle of A yielding) where I don't think I would want A to wake quite yet.